src/common/resources/group.ts

```diff
@@ -0,0 +1,96 @@
+import { Application, Entity, Group, Policy, User } from 'common/typedefs';
+import { MaskLevel, Permission } from 'common/typedefs/Permission';
+import { IListParams, IListResponse } from 'common/typedefs/Resource';
+import {
+  createGroup,
+  deleteGroup,
+  getGroup,
+  getGroups,
+  updateGroup,
+  addApplicationToGroup,
+  addGroupPermissionToPolicy,
+  addGroupToUser,
+  removeApplicationFromGroup,
+  removeGroupPermissionFromPolicy,
+  removeGroupFromUser,
+} from 'services';
+import { GroupWithMask, PolicyWithMask } from 'services/types';
+
+interface GroupResourceInterface {
+  createItem: ({ item }: { item: Partial<Group> }) => Promise<Group>;
```

Comment:
I don't think `Partial<Group>` is sufficient type here... there are a specific set of required fields that must be provided to call this method.

The type you want for this maybe looks like:
```
type CreateGroupInput = {name: string} & Partial<Group>;
```

specify the required fields and their types then give it an Intersection (&) with the Partial Group to include all the remaining fields as optional.

Reply:
Oh i see what you mean, good idea - will do!